runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -1002,6 +1007,36 @@ private void onClientInitialBegin(
 
         }
 
+        private void onClientInitialFlush(
+            FlushFW flush)
+        {
+            final OctetsFW extension = flush.extension();
+            final ExtensionFW flushEx = extension.get(extensionRO::tryWrap);
+
+            final KafkaFlushExFW kafkaFlushEx = flushEx != null && flushEx.typeId() == kafkaTypeId ?
+                extension.get(kafkaFlushExRO::tryWrap) : null;
+
+            assert kafkaFlushEx != null;
+            assert kafkaFlushEx.kind() == KafkaFlushExFW.KIND_FETCH;
+            final KafkaFetchFlushExFW kafkaFetchFlush = kafkaFlushEx.fetch();
+            final Array32FW<KafkaFilterFW> filters = kafkaFetchFlush.filters();
+            final KafkaEvaluation evaluation = kafkaFetchFlush.evaluation().get();
+            final KafkaFilterCondition condition = cursorFactory.asCondition(filters, evaluation);
+
+            newCursor = cursorFactory.newCursor(condition, deltaType);
+            Node segmentNode = group.partition.seekNotAfter(initialOffset);
+            if (segmentNode.sentinel())
+            {
+                segmentNode = segmentNode.next();
+            }
+
+            newCursor.init(segmentNode, currentCursor.offset, currentCursor.latestOffset);
```

Comment:
Suggest instead adding a new `init` method that takes the `KafkaCacheCursor` and uses the fields necessary to initialize, such as `segmentNode`, `offset` and `latestOffset`.